### ISSUE TITLE
Bug in rule “headers.h1-title”: [WHAT]

### ISSUE THREAD

**Issue by wseltzer** [w3c/specberus#839]
This rule should ignore whitespace: 

Title is: Web Authentication: An API for accessing Public Key Credentials -- Level 2

while the H1 inserts line-breaks: 
Web Authentication:
An API for accessing Public Key Credentials
Level 2

Found [while checking `https://www.w3.org/TR/2019/WD-webauthn-2-20190504/`](https://www.w3.org/pubrules/?url=https%3A%2F%2Fwww.w3.org%2FTR%2F2019%2FWD-webauthn-2-20190504%2F&profile=WD&validation=simple-validation&noRecTrack=false&informativeOnly=false&echidnaReady=false&patentPolicy=pp2004).

**Comment by deniak** [comment]
The [rule](https://github.com/w3c/specberus/blob/master/lib/rules/headers/h1-title.js) compares the `<title>` and the `<h1>`. Even if we were to ignore whitespaces and line breaks, the hyphen in the `<title>` would still be problematic.

**Comment by wseltzer** [comment]
Can the rule be updated to ignore that too?

**Comment by deniak** [comment]
That's doable, but we first need to detail what's allowed by the rule as the different generators try to achieve the same thing but with different structures, e.g. https://github.com/w3c/specberus/issues/838.
Here's a proposal:
- the rule strips all the ` - ` (including the surrounding whitespaces) in the `<title>`;
- in the headers, add the level in a `<h2 id=subtitle>` following the `<h1>` and the rule concatenates these 2 elements.

This way, we can have something like:
```html
<title>Foo bar - level N</title>
...
<h1>Foo bar</h1>
<h2 id=subtitle>level N</h2>
```

WDYT?

/cc @plh, @marcoscaceres, @tabatkins

**Comment by tabatkins** [comment]
I don't understand - the current common usage of subtitle across W3C specs has it contain the spec status and date, like "Editors Draft, 30 May 2019" <https://drafts.csswg.org/css-values/>. That matches what WebAuthn is currently doing, too.

**Comment by marcoscaceres** [comment]
(not my doing, promise!) ReSpec has a legacy feature that allows for subtitle:
https://w3c.github.io/manifest/?subtitle="Hi%20Tab"

**Comment by tabatkins** [comment]
Ah, I missed that the `<h2>Status, Date</h2>` element in ReSpec specs does *not* use `id=subtitle`. It does in all current Bikeshedded W3C specs, because that was the pattern used in the CSSWG's templates pre-Bikeshed (back when we were using Bert's preprocessor), and other boilerplates have generally copypasted from the CSSWG's original boilerplate.

That can be changed if necessary, it's just some edits to the boilerplate files.

**Comment by tabatkins** [comment]
> because that was the pattern used in the CSSWG's templates pre-Bikeshed

...huh, that's not true. [The old module template](https://drafts.csswg.org/css-module/) doesn't use it, and that's what I used to generate the CSSWG's original Bikeshed boilerplates.

It's not present in the sole remaining legacy CSSWG spec, Fonts 3, either.  Where did I get this from???

**Comment by deniak** [comment]
Since the use of the `subtitle` identifier isn't consistent, we could introduce another one that both respec and bikeshed could use to mention the level.

**Comment by tabatkins** [comment]
Turns out I added it for no apparent reason back in 2013: https://github.com/tabatkins/bikeshed/commit/5a0061a0cc634d5a052b0c876616276e0afd3a25

Anyway, `class=subtitle` is not used by Bikeshed, but is used by ReSpec. Feel free to key off of that if you want to do this.

**Comment by jennyliang220** [comment]
Fixed by #1044, now check for [Web Authentication: An API for accessing Public Key Credentials: Level 2](https://www.w3.org/pubrules/?url=https%3A%2F%2Fwww.w3.org%2FTR%2F2020%2FWD-webauthn-2-20201116%2F&profile=WD&validation=simple-validation&noRecTrack=false&informativeOnly=false&echidnaReady=false&patentPolicy=pp2004) shows no error.
